test/e2e/utils/pod.go

```diff
@@ -0,0 +1,244 @@
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubectl/pkg/util/podutils"
+)
+
+// NFDMasterPod provide NFD master pod definition
+func NFDMasterPod(image string, onMasterNode bool) *v1.Pod {
+	p := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "nfd-master-",
+			Labels:       map[string]string{"name": "nfd-master-e2e"},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:            "node-feature-discovery",
+					Image:           image,
+					ImagePullPolicy: v1.PullAlways,
+					Command:         []string{"nfd-master"},
+					Env: []v1.EnvVar{
+						{
+							Name: "NODE_NAME",
+							ValueFrom: &v1.EnvVarSource{
+								FieldRef: &v1.ObjectFieldSelector{
+									FieldPath: "spec.nodeName",
+								},
+							},
+						},
+					},
+				},
+			},
+			ServiceAccountName: "nfd-master-e2e",
+			RestartPolicy:      v1.RestartPolicyNever,
+		},
+	}
+	if onMasterNode {
+		p.Spec.NodeSelector = map[string]string{"node-role.kubernetes.io/master": ""}
+		p.Spec.Tolerations = []v1.Toleration{
+			{
+				Key:      "node-role.kubernetes.io/master",
+				Operator: v1.TolerationOpEqual,
+				Value:    "",
+				Effect:   v1.TaintEffectNoSchedule,
+			},
+		}
+	}
+	return p
+}
+
+// NFDWorkerPod provides NFD worker pod definition
+func NFDWorkerPod(image string, extraArgs []string) *v1.Pod {
+	p := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nfd-worker-" + string(uuid.NewUUID()),
+		},
+		Spec: *nfdWorkerPodSpec(image, extraArgs),
+	}
+
+	p.Spec.RestartPolicy = v1.RestartPolicyNever
+
+	return p
+}
+
+// NFDWorkerDaemonSet provides the NFD daemon set worker definition
+func NFDWorkerDaemonSet(image string, extraArgs []string) *appsv1.DaemonSet {
+	podSpec := nfdWorkerPodSpec(image, extraArgs)
+	return newDaemonSet("nfd-worker", podSpec)
+}
+
+// newDaemonSet provide the new daemon set
+func newDaemonSet(name string, podSpec *v1.PodSpec) *appsv1.DaemonSet {
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name + "-" + string(uuid.NewUUID()),
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"name": name},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"name": name},
+				},
+				Spec: *podSpec,
+			},
+			MinReadySeconds: 5,
+		},
+	}
+}
+
+func nfdWorkerPodSpec(image string, extraArgs []string) *v1.PodSpec {
+	return &v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Name:            "node-feature-discovery",
+				Image:           image,
+				ImagePullPolicy: v1.PullAlways,
+				Command:         []string{"nfd-worker"},
+				Args:            append([]string{"--server=nfd-master-e2e:8080"}, extraArgs...),
```

Comment:
Has been changed since this patch was initially written

```suggestion
				Args:            append([]string{"-server=nfd-master-e2e:8080"}, extraArgs...),
```

Reply:
ah, missed! fixing, thanks.